src/System.Resources.ReaderWriter/src/System/Resources/ResourceReader.cs

```diff
@@ -90,15 +86,96 @@ private unsafe int GetNamePosition(int index)
             }
             return r;
         }
-        public IDictionaryEnumerator GetEnumerator()
+
+        public Enumerator GetEnumerator()
         {
             if (_store == null)
                 throw new InvalidOperationException(SR.ResourceReaderIsClosed);
-            return new ResourceEnumerator(this);
+            return new Enumerator(this);
         }
 
+        public struct Enumerator
+        {
+            private const int ENUM_DONE = Int32.MinValue;
+            private const int ENUM_NOT_STARTED = -1;
+
+            private ResourceReader _reader;
+            private bool _currentIsValid;
+            private int _currentName;
+
+            internal Enumerator(ResourceReader reader)
+            {
+                _currentIsValid = false;
+                _currentName = ENUM_NOT_STARTED;
+                _reader = reader;
+            }
+
+            public bool MoveNext()
+            {
+                if (_currentName == _reader._numResources - 1 || _currentName == ENUM_DONE)
+                {
+                    _currentIsValid = false;
+                    _currentName = ENUM_DONE;
+                    return false;
+                }
+                _currentIsValid = true;
+                _currentName++;
+                return true;
+            }
+
+            public KeyValuePair<string, object> Current
+            {
+                get
+                {
+                    if (_currentName == ENUM_DONE) throw new InvalidOperationException(SR.InvalidOperation_EnumEnded);
+                    if (!_currentIsValid) throw new InvalidOperationException(SR.InvalidOperation_EnumNotStarted);
+
+                    String key;
+                    Object value = null;
+                    int _dataPosition;
```

Comment:
This is actually mainly copy-paste from the old enumerator. But I will change it.